src/components/themed/MiniButton.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+
+import * as React from 'react'
+import { ActivityIndicator, Text, TouchableOpacity } from 'react-native'
+import { cacheStyles } from 'react-native-patina'
+
+import { useState } from '../../types/reactHooks.js'
+import { fixSides, mapSides, sidesToMargin } from '../../util/sides.js'
+import { type Theme, useTheme } from '../services/ThemeContext.js'
+
+type Props = {|
+  // Called when the user presses the button.
+  // If the callback returns a promise, the button will disable itself
+  // and show a spinner until the promise resolves.
+  onPress?: () => void | Promise<void>,
+
+  // Whether to center the button or stretch to fill the screen.
+  // Defaults to 'auto', letting the parent component be in charge:
+  alignSelf?: 'auto' | 'stretch' | 'center',
+
+  // True to dim the button & prevent interactions:
+  disabled?: boolean,
+
+  // If this is set, the component will insert a text node before the other children:
+  label?: string,
+
+  // The gap around the button. Takes 0-4 numbers (top, right, bottom, left),
+  // using the same logic as the web `margin` property. Defaults to 0.
+  marginRem?: number[] | number
+|}
+
+/**
+ * A small button used for max sends and such.
+ * Visually, this happens to look like a 5/8 scale secondary button.
+ */
+export function MiniButton(props: Props) {
+  const { alignSelf = 'auto', disabled = false, label, marginRem, onPress } = props
+
+  // `onPress` promise logic:
+  const [pending, setPending] = useState(false)
+  const handlePress = () => {
+    if (onPress == null || pending) return
+    const out = onPress()
+    if (out != null && typeof out.then === 'function') {
+      setPending(true)
+      const onDone = () => setPending(false)
+      out.then(onDone, onDone)
+    }
+  }
+
+  // Styles:
+  const theme = useTheme()
+  const styles = getStyles(theme)
+  const dynamicStyles = {
+    alignSelf,
+    opacity: disabled || pending ? 0.7 : 1,
+    ...sidesToMargin(mapSides(fixSides(marginRem, 0), theme.rem))
+  }
```

Comment:
In an effort to make it a little DRY, could we move some of this shared functionality into hooks in order to use the in both `MainButton` and `MiniButton`? This makes a "has a" inheritance model rather than an "is a" model. If this is feasible, then it's worth exploring.

Reply:
The styles are unique, and PRY is the right approach. The button spinning logic could maybe be moved into a hook, so I'll give that a shot.